DetectorDescription/Core/src/DDCompactView.cc

```diff
@@ -100,6 +110,20 @@ void DDCompactView::lockdown() {
   DDSpecifics::StoreT::instance().swap(specStore_);
   DDRotation::StoreT::instance().swap(rotStore_);
 
+  //Deal with DDVectors
+  DDVector::iterator<DDVector> vit;
+  DDVector::iterator<DDVector> ved(DDVector::end());
+
+  for (; vit != ved; ++vit) {
+    if (vit->isDefined().second) {
+      DDName vname(vit->name());
+      vectors_.emplace(vname.name(), vit->values());
```

Comment:
@Dr15Jones - the map will have a vector with a name and without its namespace. The lookup in the user code is done by name. It is possible, however, to define objects with the same name in the different namespaces. 

Perhaps, to throw an exception or generate an error would be appropriate? Something like that "a re-definition of a vector <name> in the namespace <name> is ignored"

Reply:
@ianna I was just trying to have the exact same behavior as the original code. That code is

https://github.com/cms-sw/cmssw/blob/3050874051b66e90aff20ec2c6e18e1ae3604944/DetectorDescription/Core/interface/DDVectorGetter.h#L28-L42

Where the first entry in the map is the one that is obtained (which is also how the new code works). 

The present code doesn't generate an error for multiple possible definitions and we are working on replacing this code with DD4Hep anyway I'd prefer not to extend the functionality at this point. If you really want, I could add a LogError or LogWarning but that will just only uncover a problem we've ignored for many years :).

Reply:
fair enough :-)